src/Relude/List.hs

```diff
@@ -13,7 +11,39 @@ Utility functions to work with lists.
 module Relude.List
        ( module Relude.List.NonEmpty
        , module Relude.List.Reexport
+       , (!!?)
        ) where
 
+
+import Relude.Base ((<))
+import Relude.Bool (otherwise)
 import Relude.List.NonEmpty
 import Relude.List.Reexport
+import Relude.Monad (Maybe (..))
+import Relude.Numeric (Int, (-))
+
+
+-- $setup
+-- >>> import Relude
+
+{- | Safer version of 'Relude.Unsafe.!!', returns a Maybe.
+get element from list using index value starting from `0`.
+
+>>> [] !!? 0
+Nothing
+
+>>> ["a", "b", "c"] !!? 3
+Nothing
+
+>>> ["a", "b", "c"] !!? 2
+Just "c"
+-}
+infixl 9 !!?
+(!!?) :: [a] -> Int -> Maybe a
+(!!?) xs i
+  | i < 0     = Nothing
+  | otherwise = f i xs
+  where f 0 (x:_)  = Just x
+        f j (_:ys) = f (j - 1) ys
+        f _ []     = Nothing
```

Comment:
Function definition should go on the line after `where`, should have type signature and be indented with 4 spaces, like this:

```haskell
  where
    f :: ...
    f ...
```

Also, let's rename `f` to `go` for more idiomatic naming